### PR TITLE
fix(CodeSnippet): horizontal scrollbars missing

### PR DESCRIPTION
**Pull request by guigueb**
This PR is in draft mode.
- #7574 requires UX input

This PR is not the solution for #7574.
This PR is to discuss the issues.
And in the case of fixing the horizontal scroll bars, to demonstrate the problems that need UX input.

```
Please note that scrollbars have a different look and feel on different platforms.
This issue looks different on Mac and windows.
```

This is the current layout of CodeSnippet...

```
<div class="bx--snippet-container" style={{overflow-y: auto}}>
   <code>
      <pre style={{overflow-x: auto}}>
      </pre>
   </code>
</div>
```

The vertical scrollbar(overflow-y) is available in both closed and expanded mode when the data exceeds the height.
UX is needed for this because the scroll bar collides with the copy and show button.
See attachment (vertical scrollbar issues.mov).

The horizontal scrollbar(overflow-x) are available when wrapText = F and the data exceeds the width.
The current problem with the horizontal scrollbar is that it is on the `<pre>` tag.
Meaning: the scroll bar is part of the vertically scrolled content and is not visible until you scroll to the bottom of the data.
ie: You can horizontally scroll without ever seeing the scrollbar.
See attachment (horizontal scrollbar issues.mov).

The horizontal scrollbar needs to be on the `<div class="bx--snippet-container">` tag, the same as the vertical scrollbar.
This PR demos moving it to that location.
The problem with doing that is that the horizontal scrollbar then collides with the show button (same issue the vertical scrollbar has).
This is the UX that is needed to address this.
See attachment (horizontal issue when on div.mov).

### CODE REVIEW

**Review by emyarod — COMMENTED** [review]
requesting visual review cc @carbon-design-system/design

**Comment by johnbister** [comment]
@guigueb @emyarod looking into this today.

**Comment by johnbister** [comment]
@guigueb  @emyarod The scrollbars are intentionally left out in this view because the 'show more' button is activated if the snippet is greater than 15 lines and it wraps by default. To have both a scrollbar and a 'show more' would feel a little redundant.  Thanks for submitting this, but we're comfortable with how this component looks for now.

**Review by johnbister — COMMENTED** [review]
The scrollbars are intentionally left out in this view because the 'show more' button is activated if the snippet is greater than 15 lines and it wraps by default. To have both a scrollbar and a 'show more' would feel a little redundant. Thanks for submitting this, but we're comfortable with how this component looks for now.

**Comment by joshblack** [comment]
Going to close this out based on the above note, feel free to comment though and I can re-open if this is a mistake!

**Comment by guigueb** [comment]
Uploading vertical scrollbar issues.mov…

**Comment by guigueb** [comment]
https://user-images.githubusercontent.com/5973642/113770946-c7014a00-96f0-11eb-9daa-72c67b53df42.mov

**Comment by guigueb** [comment]
https://user-images.githubusercontent.com/5973642/113771331-39722a00-96f1-11eb-96a4-b2563a485e51.mov

**Comment by guigueb** [comment]
@johnbister 
@joshblack
@emyarod

Please have another read of this issue, and if need be please reach out and we can talk about this.
I fear I didn't make a clear explanation of the issue.
I re-worded the description and included videos of the `existing` issues.

I did not include a video of the issues seen when the horizontal scrollbar option is in the right place - on the div - as the demo generated from this PR was deleted when the PR was closed.
It can be recreated if the PR is re-opened.
Basically you will see that the horizontal scrollbar will collide with the show button in the same way that the vertical scrollbar currently collides with the show button.

**Comment by guigueb** [comment]
https://user-images.githubusercontent.com/5973642/113896204-b7890c00-9797-11eb-8b1c-1c3b2d99f30d.mov

**Comment by joshblack** [comment]
Wanted to circle back on this PR, what was the latest on this @johnbister?? 👀

**Comment by johnbister** [comment]
@joshblack @guigueb I reached out to the design team to discuss this and #8322. Standby and thanks for your patience.
